bundles/org.openhab.binding.visualcrossing/src/main/java/org/openhab/binding/visualcrossing/internal/WeatherCrossingActions.java

```diff
@@ -0,0 +1,82 @@
+import org.openhab.binding.visualcrossing.internal.api.VisualCrossingRateException;
+import org.openhab.binding.visualcrossing.internal.api.dto.WeatherResponse;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.openhab.core.thing.binding.ThingActions;
+import org.openhab.core.thing.binding.ThingActionsScope;
+import org.openhab.core.thing.binding.ThingHandler;
+
+/**
+ * @author Martin Grzeslowski - Initial contribution
+ */
+@ThingActionsScope(name = BINDING_ID)
+@NonNullByDefault
+public class WeatherCrossingActions implements ThingActions {
+    private @Nullable VisualCrossingHandler handler;
+
+    @Override
+    public void setThingHandler(ThingHandler thingHandler) {
+        this.handler = (VisualCrossingHandler) thingHandler;
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return handler;
+    }
+
+    @RuleAction(label = "@text/action.timeline.label", description = "@text/action.timeline.description")
+    @Nullable
+    public WeatherResponse timeline(
+            @ActionInput(name = "location", label = "@text/action.label.location", description = "@text/action.description.location") @Nullable String location,
+            @ActionInput(name = "unitGroup", label = "@text/action.label.unitGroup", description = "@text/action.description.unitGroup") @Nullable UnitGroup unitGroup,
+            @ActionInput(name = "lang", label = "@text/action.label.lang", description = "@text/action.description.lang") @Nullable String lang,
+            @ActionInput(name = "dateFrom", label = "@text/action.label.dateFrom", description = "@text/action.description.dateFrom") @Nullable String dateFrom,
+            @ActionInput(name = "dateTo", label = "@text/action.label.dateTo", description = "@text/action.description.dateTo") @Nullable String dateTo)
+            throws VisualCrossingAuthException, VisualCrossingApiException, VisualCrossingRateException {
+        var localHandler = handler;
+        if (localHandler == null) {
+            return null;
+        }
+        return localHandler.timeline(location, unitGroup, lang, dateFrom, dateTo);
+    }
+
+    public static void timeline(@Nullable ThingActions actions, @Nullable String location,
+            @Nullable UnitGroup unitGroup, @Nullable String lang, @Nullable String dateFrom, @Nullable String dateTo)
+            throws VisualCrossingAuthException, VisualCrossingApiException, VisualCrossingRateException {
+        ((VisualCrossingHandler) requireNonNull(actions)).timeline(location, unitGroup, lang, dateFrom, dateTo);
+    }
+
+    @RuleAction(label = "@text/action.timeline.label", description = "@text/action.timeline.description")
+    @Nullable
+    public WeatherResponse timeline()
+            throws VisualCrossingAuthException, VisualCrossingApiException, VisualCrossingRateException {
+        return timeline(null, null, null, null, null);
```

Comment:
What is the purpose of this Action?

Reply:
What is the purpose of adding this parameterless method that is only calling `timeline` with all null variables ?

Reply:
What I see OH is often providing 2 flavors of methods: all parameters, required parameters and because there are no required parameters I'm passing all nulls.

Look at `say` API from core OH - there are few flavors of the same method:

 - say(Object text) : says a given text with the default voice
 - say(Object text, PercentType volume) : says a given text with the default voice and the given volume
 - say(Object text, String voice) : says a given text with a given voice
 - say(Object text, String voice, PercentType volume) : says a given text with a given voice and the given volume
 - say(Object text, String voice, String sink) : says a given text with a given voice through the given sink
 - say(Object text, String voice, String sink, PercentType volume) : says a given text with a given voice and the given volume through the given sink

Reply:
But the question remains the same. What is the purpose of this method, calling `timeline()` does nothing does it?
Or what am i missing?